corehq/ex-submodules/couchforms/getters.py

```diff
@@ -54,6 +69,23 @@ def get_instance_and_attachment(request):
     return instance, attachments
 
 
+def _valid_xml_extension(file):
+    return _valid_file_extension(file, PERMITTED_FORM_SUBMISSION_FILE_EXTENSIONS)
+
+
+def _valid_file_extension(file, permitted_extensions):
+    # validate that file has only one extension and that is a permitted one
+    try:
+        file_extension = file.name.split('.', 1)[1]
+    except IndexError:
+        return False
```

Comment:
`rsplit` will split on the last `.` in the filename rather than the first, which will be more reliable
```suggestion
    if "." not in file.name:
        return False
    file_extension = file.name.rsplit('.', 1)[-1]
```

Example results:
```py
>>> names = [".abc", "a.bc", "a.b.c", "abc."]
>>> for name in names:
...     print(name, "->", repr(name.rsplit(".", 1)[-1]))
... 
.abc -> 'abc'
a.bc -> 'bc'
a.b.c -> 'c'
abc. -> ''
```

Reply:
thanks for the suggestion there @millerdev .
Another thing that we need to consider was multiple extensions, so `image.php.jpeg` is also an invalid file name. Checking on the first "." handled that scenario as well, though implicitly which is not good.
I can either add a comment so that that is clear or write some code to check on number of occurrences of ".".

Reply:
`image.php.jpeg` seems like a perfectly valid (jpeg image) filename to me. Why is that not good?

Reply:
> a filename ideally should not have two extensions right?

File extensions are a convention, and how they are interpreted is dependent on the program that is processing the file. Typically only the part of the filename after the last period counts as the extension, and there are no restrictions on putting period characters or almost any other character in the part of the filename before the extension. There are cases where multiple extensions are used by convention. For example compressed a tar images (.tar.gz). I'm surprised this is a real concern.

Reply:
that makes sense. the example of `.tar.gz` is definitely a valid one. At this point we are talking about single files and zipped files are not supported and I can't think of any other example. 
Plus, I think it would be very very rare for users to upload files with multiple period signs in the name.
Let me know if you think we should find out more about this from audit team.

Reply:
There will likely be a support burden if an unknowing user tries to upload files that have periods in the filename (this is a perfectly valid thing to do). I think it would be best to ignore all but the final period + extension to avoid potential confusion.